### DISCUSSION TITLE
Индивидуальный селект только при добавлении и изменении

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#273] [Как это сделать?] [answered]
Есть большие таблицы, в которых выбор значений зависит от других значений в таблице.

Вопрос: можно ли сделать так, чтобы расчет индивидуального селекта производился только при добавлении или изменении строки, а все остальное время селект при обновлении таблицы не пересчитывался?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Когда стоит индивидуальный селект, то он будет рассчитываться для каждой строки при каждом обновлении.

Иногда в циклах внутри используют индивидуальный в хедере, а потом достают его в таблицу циклов ярлыком — это плохая идея, лучше сделать в таблице циклов это поле нерадактируемым и сделать там неиндивидуальный селект, а значение взять через `select`.

Если речь идет про строчную часть, то можно облегчить расчет индивидуального селекта сделав генерацию отображения для существующей строки.

Строка добавления определяется по отсутствии у нее `id`:

```
=: if(condition: #id = ""; then: $add; else: $manual)

add: selectRowListForSelect(....)
manual: rowCreate(field: "value" = $#nfv; field: "title" = #store_title)

```

В поле `store_title` записывать отображение кодом при добавлении, оно по `sort` может стоять после поля с Селектом.

При таком подходе поле нельзя редактировать стандартными средствами, но можно сделать linkToInputSelect по двойному клику.

**Reply by erpic-ru** [reply]
Небольшой фикс: selectRowListForSelect ждет массив, поэтому нужно его создать с одной строкой. Ну и поле store_title тоже стоит обновлять при редактировании через linkToInputSelect
```

=: if(condition: #id = ""; then: $add; else: $manual)

add: selectRowListForSelect(....)
manual: listCreate(item: $row)
   row:rowCreate(field: "value" = $#nfv; field: "title" = #store_title)
```
